pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/rawingress_reconciler.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2021 kubeflow.org.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	v1beta1api "github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//NativeReconciler is the struct of Native K8S Object
+type RawIngressReconciler struct {
+	client          client.Client
+	scheme          *runtime.Scheme
+	Ingress         *netv1.Ingress
+	componentExt    *v1beta1.ComponentExtensionSpec
+	componentStatus v1beta1.ComponentStatusSpec
+}
+
+func NewRawIngressReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *RawIngressReconciler {
+	return &RawIngressReconciler{
+		client:          client,
+		scheme:          scheme,
+		Ingress:         createRawIngress(client, componentMeta, componentExt, podSpec, componentStatus),
+		componentExt:    componentExt,
+		componentStatus: componentStatus,
+	}
+}
+
+//GenerateIngressHost return the config domain in configmap.IngressDomain
+func GenerateIngressHost(client client.Client, metadata metav1.ObjectMeta) string {
+	ingressConfig, err := v1beta1api.NewIngressConfig(client)
+	if err != nil {
```

Comment:
It could be configuration parsing error also, prefer not masking those issues. We can set the default value in config instead.